src/main/java/coding/exercises/programmers/ExampleProgrammerService.java

```diff
@@ -0,0 +1,52 @@
+package coding.exercises.programmers;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class ExampleProgrammerService implements ProgrammerService {
+
+  private Collection<Programmer> programmers;
+
+  public ExampleProgrammerService(Collection<Programmer> programmers) {
+    this.programmers = programmers;
+  }
+
+  @Override
+  public Optional<Programmer> findSomeoneWhoKnowsLanguage(String language) {
+    return programmers.stream()
+        .filter(p -> p.getLanguages().contains(language))
+        .findAny();
+  }
+
+  @Override
+  public Set<String> allLanguagesCoveredByProgrammers() {
+    return programmers.stream()
+        .map(Programmer::getLanguages)
+        .flatMap(Collection::stream)
+        .collect(Collectors.toSet());
+  }
+
+  @Override
+  public Integer sumOfAllSalaries() {
+    return programmers.stream()
+        .map(Programmer::getSalary)
+        .reduce(0, Integer::sum);
```

Comment:
There is also a different technique using the `Collectors`. I don't know how to add it here :-)
```java
return programmers.stream()
  .map(Programmer::getSalary)
  .collect(Collectors.summingInt(Integer::intValue));
```